packages/core/src/application.ts

```diff
@@ -280,141 +32,89 @@ export class Application extends Context {
    * modifications.
    *
    * ```ts
-   * @spec(apiSpec)
    * class MyController {
    * }
    * app.controller(MyController).lock();
    * ```
    */
-  controller(controllerCtor: ControllerClass): Binding {
+  controller(controllerCtor: Constructor<{}>): Binding {
```

Comment:
Can we bring back `ControllerClass` please?

I think it may make sense to describe the expected controller interface too, something like the following:

```
interface Controller {
  [key: string]: (...args[]: any): ValueOrPromise<any>;
}

type ControllerClass = Constructor<Controller>; // can we use `typeof Controller`?
```